x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/MaintenanceModeUpdateTask.java

```diff
@@ -28,16 +28,23 @@ OperationMode getOperationMode() {
     @Override
     public ClusterState execute(ClusterState currentState) {
         IndexLifecycleMetadata currentMetadata = currentState.metaData().custom(IndexLifecycleMetadata.TYPE);
-
-
-        if (currentMetadata.getMaintenanceMode().isValidChange(mode) == false) {
+        if (currentMetadata != null && currentMetadata.getMaintenanceMode().isValidChange(mode) == false) {
             return currentState;
+        } else if (currentMetadata == null) {
+            currentMetadata = IndexLifecycleMetadata.EMPTY;
+        }
+
+        final OperationMode newMode;
+        if (currentMetadata.getMaintenanceMode().isValidChange(mode)) {
+            newMode = mode;
+        } else {
+            newMode = currentMetadata.getMaintenanceMode();
```

Comment:
Can this be any cleaner? We seem to be checking things twice. Maybe we could check for a nell metadata first and then check if the change is valid?